### DISCUSSION TITLE
Removing HTTPS requirement for Amateur Radio deployment?

### DISCUSSION THREAD

**Discussion by clintonnichols** [mastodon/mastodon#18581] [General]
Amateur radio has various digital modes, in particular AREDN mesh networks, where I think Mastodon would be useful. Use of encryption is generally not permitted on amateur radio. I'm looking for guidance on how to remove the dependency on encryption in Mastodon, if possible.

Thank You!

**Comment by nightpool** [comment]
This would be a pretty involved change, since it would require removing all concept of private and direct posts, since there would be no way to transmit those posts confidentially. I'm not familiar enough with AREDN to say whether it would even be compatible with Mastodon, considering that it relies heavily on the request-response protocol provided by HTTP and the global name resolution provided by DNS.

**Reply by clintonnichols** [reply]
Thank you. I appreciate the response and if the use of HTTPS is required for the platform to function properly then I agree that it wouldn't be a good fit for an AREDN network.  FYI, AREDN is a mesh TCP/ip network available to amateur radio operators using off-the-shelf wireless routers running modified firmware on amateur radio frequencies. Encryption is rarely allowed by the FCC on amateur radio bands. Running what is essentially a private WAN where network encryption is prohibited does have its challenges.
